importer.go

```diff
@@ -155,41 +183,12 @@ func (i *Importer) fetcher() URLFetcher {
 // to this importer, we need to preserve that prefix.
 // TODO(camh): Create a "netpath" package with these preservation semantics.
 func preserveNetRoot(orig, cleaned string) string {
-	if len(orig) > 1 && orig[0] == '/' && orig[1] == '/' {
+	if isNetpath(orig) {
 		return "/" + cleaned
 	}
 	return cleaned
 }
 
-func addScheme(imp string) (string, error) {
-	if strings.HasPrefix(imp, "//") {
-		return "https:" + imp, nil
-	}
-
-	abs, err := filepath.Abs(imp)
-	if err != nil {
-		return "", err
-	}
-	return "file://" + abs, nil
-}
-
-func fetch(u string, f URLFetcher) (jsonnet.Contents, error) {
-	resp, err := f.Get(u)
-	if err != nil {
-		return noContent, err
-	}
-	defer resp.Body.Close()
-
-	if resp.StatusCode == http.StatusNotFound {
-		return noContent, nil
-	} else if resp.StatusCode != http.StatusOK {
-		return noContent, fmt.Errorf("could not fetch %#v: %s", u, resp.Status)
-	}
-
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return noContent, err
-	}
-
-	return jsonnet.MakeContents(string(b)), nil
+func isNetpath(path string) bool {
+	return len(path) > 1 && path[0] == '/' && path[1] == '/'
```

Comment:
is `//` a valid netpath? is it worth while considering further validation?

Reply:
it is not valid, but it will fail when you try to fetch it.